src/IO/Archives/LibArchiveReader.cpp

```diff
@@ -147,15 +186,42 @@ class LibArchiveReader::Handle
         file_info.reset();
     }
 
-    static struct archive * open(const String & path_to_archive)
+    Archive openWithReader(StreamInfo * read_stream_)
+    {
+        auto * archive = archive_read_new();
+        try
+        {
+            archive_read_support_filter_all(archive);
```

Comment:
let's apply changes from this PR https://github.com/ClickHouse/ClickHouse/pull/60203/files
So we avoid some vulnerabilities

Reply:
When it gets merged I will rebase and merge conflicts.

Reply:
please apply changes from this PR as it's merged